### PR TITLE
Add survey spec

### PR DESCRIPTION
**Pull request by syncrou**
@bdunne - Please review

### CODE REVIEW

File: spec/support/factory_helper.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
 module AnsibleTowerClient
   module FactoryHelper
     def self.stringify_attribute_keys(hash)
-      [:description, :extra_vars, :id, :instance_id, :inventory, :name, :results, :type, :url].each do |attribute|
+      [:description, :extra_vars, :id, :instance_id, :inventory, :name, :results,
```
Comment (bdunne) [inline spec/support/factory_helper.rb]: If the array needs to be wrapped, it should probably be in a constant at the top.

File: spec/factories/responses.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@
     type       { AnsibleTowerClient::FactoryHelper.underscore_string(klass.namespace.last) }
     name       { "#{type}-#{id}" }
     url        { "/api/v1/#{klass.endpoint}/#{id}/" }
+    related    { {"survey_spec" => ["blah blah"]} }
```
Comment (bdunne) [inline spec/factories/responses.rb]: Is it actually an array of strings?  Usually the "related" are links to other collections, right?

**Comment by syncrou** [comment]
@bdunne - Made the requested changes.  Please review.

File: spec/job_template_spec.rb [hunk 3]
```diff
@@ -31,4 +31,15 @@
...
   end
+
+  describe '#survey_spec' do
+    let(:get) { instance_double("Faraday::Result", :body => instance.to_json) }
+
+    it "returns a survey spec if it exists" do
+      AnsibleTowerClient::Api.instance_variable_set(:@instance, api_connection)
+      survey = described_class.new(instance).survey_spec
+      expect(survey).to be_a Hash
+      expect(survey['related']['survey_spec']).to eq 'example.com/api'
```
Comment (bdunne) [inline spec/job_template_spec.rb]: I think this 'related' is wrong.  I think your get is returning a job_template rather than the survey response.

**Comment by syncrou** [comment]
@bdunne - Pushed changes including if a survey_spec is nil

**Comment by bdunne** [comment]
:+1:

**Pull request merged**